File: app/src/main/java/com/thibaudperso/sonycamera/timelapse/ui/settings/SettingsActivity.java
package com.thibaudperso.sonycamera.timelapse.ui.settings;

import android.support.v4.app.Fragment;

import com.thibaudperso.sonycamera.R;
import com.thibaudperso.sonycamera.timelapse.ui.SingleFragmentActivity;

public class SettingsActivity extends SingleFragmentActivity {

    @Override
    protected Fragment createFragment() {
        return new SettingsFragment();
    }

    @Override
    protected String getGuideTitle() {
        return getString(R.string.title_settings);
    }

    @Override
    public void finish() {
        super.finish();
        overridePendingTransition(R.anim.pop_enter, R.anim.pop_exit);
    }
}

File: app/src/main/java/com/thibaudperso/sonycamera/timelapse/ui/settings/SettingsActivity.kt
package com.thibaudperso.sonycamera.timelapse.ui.settings

import com.thibaudperso.sonycamera.R
import com.thibaudperso.sonycamera.timelapse.ui.SingleFragmentActivity

class SettingsActivity : SingleFragmentActivity<SettingsFragment>() {

    override fun finish() {
        super.finish()
        overridePendingTransition(R.anim.pop_enter, R.anim.pop_exit)
    }

    override fun createFragment() = SettingsFragment()

    override val guideTitle: String
        get() = getString(R.string.title_settings)
}
